src/fakeRequest.js

```diff
@@ -326,6 +353,50 @@ extend(FakeXMLHttpRequest, {
         stub.action = undefined;
         stub.functionToCall(stub, this);
         this.dispatchStub(stub);
+      },
+
+      responseStream: function(stub) {
+        if (this.readyState >= FakeXMLHttpRequest.LOADING) {
+          throw new Error("FakeXMLHttpRequest already loading or finished");
+        }
+
+        this.status = stub.status;
+        this.responseText = "";
+        this.statusText = stub.statusText || "";
+        this.responseHeaders = normalizeHeaders(stub.responseHeaders, stub.contentType);
+        this.readyState = FakeXMLHttpRequest.HEADERS_RECEIVED;
+        this.eventBus.trigger('readystatechange');
+
+        this.responseType = stub.responseType || "";
+        this.responseURL = stub.responseURL || null;
+        this.readyState = FakeXMLHttpRequest.LOADING;
+        this.eventBus.trigger('readystatechange');
+
+        var oldData = stub.streamController.data;
+        for (var i = 0; i < oldData.length; ++i) {
+          appendDataToResponse(this, oldData[i]);
+        }
+
+        if (stub.streamController.requestCancelled) {
+          finishResponseWithStatus(this, 0);
+        } else if (stub.streamController.requestFinished) {
+          finishResponseWithStatus(this, stub.status);
+        }
+
+        var request = this;
+        stub.streamController.data = {
+          push: function(text) {
+            appendDataToResponse(request, text);
+          }
+        };
+
+        stub.streamController.finishRequest = function() {
```

Comment:
This interface feels hard to discover. The user is expected to pass in an object with a certain set of properties on it, and then Jasmine-Ajax will add some methods back onto it. I think we might want the `responseStream` function (and probably the `andReturnStream` on requestStub) to return a `streamController` that the user can call some methods on to do the things they want.

Reply:
Not sure that I understood you right: do you suggest to override some properties of passed user's object into `andReturnStream` function?